lighthouse-core/audits/byte-efficiency/library-suggestions.json

```diff
@@ -0,0 +1,72 @@
+{
+  "general-purpose-date-time": [
+    "moment",
+    "dayjs",
+    "date-fns",
+    "luxon"
+  ],
+  "deep-equal": [
+    "fast-deep-equal",
+    "deep-eql",
+    "deep-equal",
+    "lodash.isequal"
+  ],
+  "schema-validation": [
+    "jsonschema",
+    "joi",
+    "ajv",
+    "superstruct",
+    "yup",
+    "validate.js"
+  ],
+  "querystring-parser": [
```

Comment:
I'm wondering if the shape of this file should just be `Array<string[]>`. Each array would be a collection of like-libraries. And then the audit would determine if the user is using anything other than the smallest one, and provide a warning with smaller alternatives. instead of just checking for the largest.

Reply:
Sorry you already jumped on this @saavannanavati but I would disagree here. If we make it an array, we're sending the message that there's no reason to use *anything other than the smallest library* which is a big stretch IMO. 

I'd like to see us be very careful about these recommendations. Inevitably it is going to be seen as "Google says I can't use X library" and there should be a very high bar for that. The cost of using the library should be *significant* and the alternative *super* obvious and easy.

`dayjs` for example has absolutely no support for timezones. If you need timezone support then Lighthouse nagging at you to switch from `luxon` to `dayjs` to save like 15 KiB *but then reimplement timezones yourself* is quite painful.

Reply:
@patrickhulce 

In this case, would we ever recommend `dayjs`? A lot of suggestions would be duplicated if we go with the non-array format, since it would be look something like this `moment: [luxon, date-fns]`,  `luxon: [date-fns]`. How do we go about determining these suggestions if not on the basis of size, because if if the issue is that a certain library is lacking a feature, then we probably shouldn't recommend it at all right?

Reply:
I'm not suggesting we never recommend put a recommendation in that's missing *a* feature of the larger library. I'm saying I would revert back to what you had before with only recommending the *set* of alternatives for particularly large library. 

In this example, *just* recommend for `moment` and suggest `luxon, date-fns, or dayjs` and if they don't need the features of luxon then hope they make the smallest choice of dayjs but don't enforce it

Reply:
Cool that makes a lot of sense, I'll change it back